Traits refactoring to more specific names instead of ElementTrait and InnerElementTrait

bboro opened the pull request:
#614 [6h]

bboro commented:
I think I prefer the latter, if we have an "InfoCard" paragraph type, and "InfoCardTrait", it makes sense that stuff for this paragraph type belongs there, and that's where we should start looking for things. What do you think?

bboro commented:
3 / 6h will continue Monday. Think I'm heading in the right direction...

amitaibu commented:
> should break it down further into "InfoCardTrait", "PeopleTrait" etc?

Yes, I agree we should break it down into smaller traits that are less abstract.

amitaibu commented on web/modules/custom/server_general/src/ElementLayoutTrait.php:
```diff
@@ -17,6 +19,7 @@
  */
 trait ElementLayoutTrait {
```
I think this file is still abstracted. Maybe break it down as well?

bboro replied:
I left it as it contains just "Layout" stuff now, such as TitleBodyAndItems etc. Where would you move these?

bboro replied:
I'm merging as is for now, we can break this down further later if you want to.

amitaibu commented on web/modules/custom/server_general/src/InnerElementLayoutTrait.php:
```diff
@@ -8,15 +8,16 @@
  *
- * A card layout can be for example a card with an image, or a card with
- * centered items. This trait should only be used by `InnerElementTrait`.
+ * An inner element can be for example a card with an image, or a search result
```
👍

amitaibu reviewed (approved):
Looks great, just a couple of comments, and MWG

Bumnp 1h to address

bboro commented:
Summary of changes:

- Break down ElementTrait and InnerElementTrait into more specific traits under `Drupal\server_general\ElementTrait\` namespace such as `CtaTrait` and `CarouselTrait`.
- Renamed `buildInnerElement...` to `buildElement...` to keep it simpler
- Moved some more "layout-y" methods from  `ElementTrait` into `ElementLayoutTrait`
- Renamed `server_theme_inner_element__...` themes to `server_theme_element__...` themes to keep it simpler.

We still have some separation of "Element" and "InnerElement" though. The idea of "InnerElement" still exists in `InnerElementLayoutTrait` and contains some layout methods for Cards etc. I decided to keep this structuring separation for now, but we can also perhaps move this to `ElementLayoutTrait` later if needed. For now I updated the comment.

TB: 5 / 6h

amitaibu commented:
> I decided to keep this structuring separation for now, but we can also perhaps move this to ElementLayoutTrait later if needed. For now I updated the comment.

Yeah, I think the fact you said it's a "private" trait is a good balance 👍
